### ISSUE TITLE
5 second delay after the end of each text generation when streaming from OpenAI API server

### ISSUE THREAD

**Issue by lksj92hs** [huggingface/chat-ui#1503]
## Bug description

There is 5 second delay after the end of each text generation when using OpenAI API server (e.g. the OpenAI servers or a local vLLM instance). Happens every time but it's especially annoying with the short answers. Tested on both Firefox and Chrome desktop browsers latest versions, as well as with OpenAI official server, OpenRouter, and a local vLLM server.

Not sure if this happens with HF endpoints (probably not or it should have been reported by now).

## Steps to reproduce

1. Start a new chat, preferably with small and fast model, with the server using OpenAI endpoint.
2. Write "Hi!"
3. The model response is streamed on the screen. After the generation is finished, the progress indicator at the bottom right side of the window is still spinning and the user input text box is disabled for exactly 5 seconds after the text generation is over.

## Context

When inspecting the HTTP requests with the browser debug console, it is obvious that the POST request with the streamed server answer takes at least 5 seconds, even after the closing `data: [DONE]` streaming message is received by the server.

I have tested both with HTTPS via ngnix proxy and direct HTTP connections to chat-ui and the issue appears in both cases.

### Specs

- **OS**: Docker image `chat-ui-db:sha-8175329` on Ubuntu 22.04 LTS
- **Browser**: Firefox and Chrome desktop versions
- **chat-ui commit**: Unsure about the exact commit that corresponds to `chat-ui-db:sha-8175329` but it's pretty recent

**NOTE:** I'm unable to test the latest versions of the docker image `chat-ui-db` because of MongoDB issue leading to inability to start MongoDB in the container with LIBC error version message in the container log. This seems to be separate issue, probably caused upstream by some misconfiguration of the MongoDB docker image

### Config

From the environment section of `docker-compose.yml`:

```
      MONGODB_URL: mongodb://localhost:27017
      OPENAI_API_KEY: ${OPENAI_API_KEY_CHAT}
      ALLOW_INSECURE_COOKIES: true
      PUBLIC_ORIGIN: "http://server01.lan:3000"
      PUBLIC_APP_NAME: ChatUI (local)
      ENABLE_ASSISTANTS: true
      TASK_MODEL: gpt-4o-mini
      USE_LOCAL_WEBSEARCH: true
      MODELS: >
              [{
                "name": "gpt-4o-mini",
                "displayName": "OpenAI/GPT-4o mini",
                "description": "GPT-4o mini (“o” for “omni”) is OpenAI's most advanced model in the small models category, and their cheapest model yet. It is multimodal (accepting text or image inputs and outputting text), has higher intelligence than GPT 3.5 Turbo but is just as fast.",
                "websiteUrl": "https://platform.openai.com/docs/models/gpt-4o-mini",
                "multimodal": true,
                "endpoints": [{ "type": "openai" }]
              }]
```
## Notes

By trying older docker images it seems that the issue was introduced around the middle of August 2024. 

After searching through the commits, I think that probably this is the commit that introduced the problem: https://github.com/huggingface/chat-ui/commit/8f10ab006b1545bc57c4b144660f4d1822fd4a7c (maybe the deleted code in `src/routes/conversation/[id]/+page.svelte`)

**Comment by nsarrazin** [comment]
Hi thanks for the report! The issue was fixed in the latest docker images already. What os/CPU are you using? 

Trying to pinpoint why you're getting errors using the latest docker image

**Comment by lksj92hs** [comment]
Thank you for the response!

I'm using Ubuntu Server 22.04.4 LTS (GNU/Linux 5.15.0-119-generic x86_64), docker version is:  27.2.0, build 3ab4256, CPU: Ryzen 7 7800X3D. I'm aware of the MongoDB issue with CPUs without AVX support but that is not the case here, as the slightly older docker images (e.g. a month or so work fine on this machine.

I haven't find any recent changes in the Dockerfile in the project, so I'm at a loss why MongoDB is not working in the `chat-ui-db` image. However the image https://github.com/huggingface/chat-ui/pkgs/container/chat-ui-db/273387685?tag=sha-8175329 from less than a month ago works fine.

**Comment by nsarrazin** [comment]
Looks like I can reproduce the issue! Working on a fix, thanks for investigating :rocket:

**Comment by nsarrazin** [comment]
https://github.com/huggingface/chat-ui/pull/1504 this PR fixed it for me locally. will merge & deploy so it can build the new images

**Comment by nsarrazin** [comment]
Deploying here: https://github.com/huggingface/chat-ui/actions/runs/11158808785 

once it's done let me know if the latest image works!

**Comment by lksj92hs** [comment]
Yes, it works for me too and the delay problem is gone too. Thank you for the quick fix and have a nice day!

I'm closing the issue.
